src/test/lib/utils/string_utils_test.cpp

```diff
@@ -38,4 +39,17 @@ TEST_F(StringUtilsTest, trim_source_file_path) {
   EXPECT_EQ(trim_source_file_path("hello/file.cpp"), "hello/file.cpp");
 }
 
+TEST_F(StringUtilsTest, table_key_constraints_to_stream) {
+  const auto table = Table::create_dummy_table(
+      {{"a", DataType::Int, false}, {"b", DataType::Int, false}, {"c", DataType::Int, false}});
+  table->add_soft_key_constraint({{ColumnID{0}, ColumnID{1}}, KeyConstraintType::UNIQUE});
+  table->add_soft_key_constraint({{ColumnID{2}}, KeyConstraintType::PRIMARY_KEY});
+  const auto separator = std::string{" | "};
+  auto stream = std::ostringstream{};
+
+  table_key_constraints_to_stream(stream, table, separator);
```

Comment:
Can you add another check for `table_key_constraints_to_stream` where `separator` is not given?